### ISSUE TITLE
Mermaid diagrams not rendering in Notable

### ISSUE THREAD

**Issue by seank-com** [fabiospampinato/notable#313]
Only Flowchart and Sequence Diagrams appear to be working.

### Repo Steps

1. Copy sample mermaid markdown from [docs](https://github.com/knsv/mermaid)
2. Paste into mermaid block in Notable
3. Press <kbd>command</kbd> + <kbd>e</kbd>

### Expected

Similar results as when you paste in the [Mermaid Live Editor](https://mermaidjs.github.io/mermaid-live-editor/)

### Current behavior

Usually a parse error similar to this:

[mermaid error: Cannot read property 'firstChild' of null]

for git Graphs there is no error, just empty space

- **OS Version**: MacOS - Mojave
- **Notable Version**: 1.1.0

**Comment by fabiospampinato** [comment]
@seank-com are you wrapping the code in a mermaid codeblock? Check an example [here](https://raw.githubusercontent.com/fabiospampinato/notable/master/resources/tutorial/notes/05%20-%20Notes.md).

**Comment by RomanKornev** [comment]
Gantt chart doesn't work like this (from docs):

    ```mermaid 
    gantt
    dateFormat  YYYY-MM-DD
    title Adding GANTT diagram to mermaid
    
    section A section
    Completed task            :done,    des1, 2014-01-06,2014-01-08
    Active task               :active,  des2, 2014-01-09, 3d
    Future task               :         des3, after des2, 5d
    Future task2               :         des4, after des3, 5d
    ```
But works like this:

    ```mermaid gantt
    dateFormat  YYYY-MM-DD
    title Adding GANTT diagram to mermaid
    
    section A section
    Completed task            :done,    des1, 2014-01-06,2014-01-08
    Active task               :active,  des2, 2014-01-09, 3d
    Future task               :         des3, after des2, 5d
    Future task2               :         des4, after des3, 5d
    ```
Seems to be a format inconsistency.

I was unable to display `Class diagram` and `Git graph` as well.

**Comment by fabiospampinato** [comment]
@RomanKornev pretty weird, just copy-pasting the first example code you provided worked for me. I'm also able to render all the samples I tried in the docs.

> I was unable to display Class diagram and Git graph as well.

Where can I find those? I don't see them mentioned in the [docs](https://mermaidjs.github.io/demos.html)

**Comment by RomanKornev** [comment]
Found the culprit:
There needs to always be a space after 
```
 ```mermaid_
````
Seems like GitHub automatically inserts a space at the end of the code line, so it worked.

**Comment by RomanKornev** [comment]
@seank-com It seems like even your example is not behaving correctly:
Compare: [Git Graph](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ2l0R3JhcGg6XG5vcHRpb25zXG57XG4gICAgXCJub2RlU3BhY2luZ1wiOiAxNTAsXG4gICAgXCJub2RlUmFkaXVzXCI6IDEwXG59XG5lbmRcbmNvbW1pdFxuYnJhbmNoIG5ld2JyYW5jaFxuY2hlY2tvdXQgbmV3YnJhbmNoXG5jb21taXRcbmNvbW1pdFxuY2hlY2tvdXQgbWFzdGVyXG5jb21taXRcbmNvbW1pdFxubWVyZ2UgbmV3YnJhbmNoXG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ)
### Docs
![git 1](https://user-images.githubusercontent.com/5807561/51502311-8efade80-1de6-11e9-957d-42d61fd5f7c9.png)
### Live editor
![2019-1-22 1-36-51](https://user-images.githubusercontent.com/5807561/51502325-a76af900-1de6-11e9-8583-d4ff29d519aa.png)

It's looping the branches multiple times. 
Even adding a character like a newline at the end and then deleting it is still adding more and more loops.
It's not regenerating the graph, it's adding more to the previous one.

**Comment by seank-com** [comment]
Awesome, I can confirm that gantt charts are fixed by adding the space.

**Comment by seank-com** [comment]
> @seank-com are you wrapping the code in a mermaid codeblock? Check an example [here](https://raw.githubusercontent.com/fabiospampinato/notable/master/resources/tutorial/notes/05%20-%20Notes.md).

Yes, that's what I meant in step #2 when I said "Paste into mermaid block in Notable"

**Comment by seank-com** [comment]
It's only the gantt diagram, class diagram and git graph that are note working. As noted in the bug, the flow chart and sequence diagram appear to be working. As @RomanKornev  pointed out, the gantt chart seems to require a space after ```mermaid to begin working. The git graph still has issues even with the space and the class diagram is broken with or without the space.

**Comment by ExpLog** [comment]
For me it's entirely the other way around. Having a space after ````mermaid` breaks it, while just having the newline works perfectly.

One thing that is odd is that on version 1.1, I could denote which type of chart I wanted with
```
```mermaid graph TD;
```
but that doesn't work now. I fixed it by changing it to
```
```mermaid
graph TD
```
without a space after mermaid, as I noted earlier, breaks it for me.

**Comment by fabiospampinato** [comment]
@ExpLog we switched from Highlight.js to Prism for highlighting, maybe that changed something in that regard.

**Comment by RomanKornev** [comment]
Everything is rendering in v1.3.0. Including ClassDiagram.
There are still some bugs remaining.
I'll outline them in #430.

**Comment by lmihalkovic** [comment]
For info, I have seen the same class diagram issue and my editor is monaco

**Comment by fabiospampinato** [comment]
I think this might be a duplicate of #621 🤔

**Comment by ramtinsoltani** [comment]
I'm using the following content in a note (without the indentation):

```
# Mermaid Test

    ```mermaid
    graph TD
      id1[A] --> id2[B]
      id2[B] --> id3[C]
      id3[C] --> id4[D]
      id4[D] --> id5[E]
      id5[E] --> id6[F]
      id6[F] --> id7[G]
      id7[G] --> id8[H]
    ```
```
When switching to preview mode, the graph is rendered for a second and then it disappears. No idea how to fix it.

**Comment by ramtinsoltani** [comment]
I'm on Windows 10 x64.

The problem exists only when switching from edit mode to preview mode. When switching from another note to the mermaid note the graph gets rendered fine.

**Comment by viktomas** [comment]
@ramtinsoltani I think you are more interested in  #621, this issue seems to track mermaid diagrams not rendering at all.

**Comment by fabiospampinato** [comment]
I can't reproduce anymore, no matter if there's a space or not after ```mermaid.

Can anybody still reproduce?
